django/db/backends/oracle/base.py

```diff
@@ -238,17 +240,69 @@ def __init__(self, *args, **kwargs):
         )
         self.features.can_return_columns_from_insert = use_returning_into
 
+    @property
+    def is_pool(self):
+        return self.settings_dict["OPTIONS"].get("pool")
+
+    @property
+    def pool(self):
+        if not self.is_pool:
+            return None
+
+        # Verify that we are not running with persistent connections
+        if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+            raise ImproperlyConfigured(
+                "Pooling doesn't support persistent connections."
+            )
+
+        if self.alias not in self._connection_pools:
+            connect_kwargs = self.get_connection_params()
+            pool_options = connect_kwargs["pool"]
+
+            if pool_options is True:
+                connect_kwargs["max"] = 4
+            else:
+                for key, val in pool_options.items():
+                    connect_kwargs[key] = val
+                if not connect_kwargs.get("max"):
+                    connect_kwargs["max"] = 4
+
+            del connect_kwargs["pool"]
+
+            pool = Database.create_pool(
+                user=self.settings_dict["USER"],
+                password=self.settings_dict["PASSWORD"],
+                dsn=dsn(self.settings_dict),
+                **connect_kwargs,
+            )
+            self._connection_pools.setdefault(self.alias, pool)
+
+        return self._connection_pools[self.alias]
+
+    def close_pool(self):
+        if self.pool:
+            self.pool.close()
+            del self._connection_pools[self.alias]
+
     def get_database_version(self):
         return self.oracle_version
 
     def get_connection_params(self):
+        # Pooling feature is only supported for oracledb
+        if self.is_pool and not is_oracledb:
+            raise ImproperlyConfigured(
+                "Pooling Not supported by cx_Oracle. Use python-oracledb instead"
```

Comment:
Change "not" to lowercase and while you're editing it, make it "isn't" to match the "doesn't" from line 255.

Or spell both out as "is not" and "does not" to match the "does not" used in django/db/backends/oracle/features.py